### PR TITLE
feat(cloudsql/postgres): update to v2 samples

### PR DESCRIPTION
**Pull request by enocom**
A peer of https://github.com/GoogleCloudPlatform/golang-samples/pull/2312.

### CODE REVIEW

File: cloudsql/postgres/database-sql/connect_connector.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	if dbUser == "" && dbIAMUser == "" {
+		log.Fatal("Wawrning: One of DB_USER or DB_IAM_USER must be defined")
+	}
+
+	dsn := fmt.Sprintf("user=%s password=%s database=%s", dbUser, dbPwd, dbName)
+	config, err := pgx.ParseConfig(dsn)
+	if err != nil {
+		return nil, err
+	}
+	config.DialFunc = func(ctx context.Context, network, instance string) (net.Conn, error) {
```
Comment (kurtisvg) [inline cloudsql/postgres/database-sql/connect_connector.go:59]: Should we be using pgx pool here?

File: cloudsql/postgres/database-sql/connect_connector.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+			d, err := cloudsqlconn.NewDialer(ctx, cloudsqlconn.WithIAMAuthN())
+			if err != nil {
+				return nil, err
+			}
+			return d.Dial(ctx, instanceConnectionName)
+			// [END cloud_sql_postgres_databasesql_auto_iam_authn]
+		}
+		if usePrivate != "" {
+			d, err := cloudsqlconn.NewDialer(ctx,
+				cloudsqlconn.WithDefaultDialOptions(cloudsqlconn.WithPrivateIP()))
```
Comment (kurtisvg) [inline cloudsql/postgres/database-sql/connect_connector.go:71]: nit: prefer line breaking between `(` and `ctx`

(or initializing the dialer on a separate line)

**Review by kurtisvg — APPROVED** [review]

File: cloudsql/postgres/database-sql/connect_connector.go (hunk 1 above)
Comment (enocom) [reply cloudsql/postgres/database-sql/connect_connector.go:59]: We're using `database/sql` as the pooling mechanism in this app, so I think we're good.

**Comment by enocom** [comment]
I've verified both public and private IP on Cloud Run, in addition to ensuring the tests pass.

**Pull request merged**